### PR TITLE
Add linear_search_reverse and linear_search_reverse_proc

### PR DESCRIPTION
**Pull request by kawaii-Code**
Surprised this was missing from the standard library, so here it is. Also added some docs for `linear_search`

### CODE REVIEW

File: core/slice/slice.odin [hunk 1]
```diff
@@ -117,6 +131,61 @@ linear_search_proc :: proc(array: $A/[]$T, f: proc(T) -> bool) -> (index: int, f
...
 }
 
+/*
+	Reverse linear search searches the given element in the given slice in O(n) time,
+	starting from the slice end.
+
+	Returns the last index at which the given element can be found in the slice,
+	or -1 if it is not present
+
+	# Examples
```
Comment (laytan) [inline core/slice/slice.odin:141]: Examples should be formatted as described here: https://github.com/odin-lang/pkg.odin-lang.org?tab=readme-ov-file#markup-rules

File: core/slice/slice.odin [hunk 2]
```diff
@@ -117,6 +131,61 @@ linear_search_proc :: proc(array: $A/[]$T, f: proc(T) -> bool) -> (index: int, f
 	return -1, false
 }
 
+/*
+	Reverse linear search searches the given element in the given slice in O(n) time,
```
Comment (laytan) [inline core/slice/slice.odin:135]: searches the -> searches for the (the case in all the added doc comments)

File: core/slice/slice.odin [hunk 3]
```diff
@@ -117,6 +131,61 @@ linear_search_proc :: proc(array: $A/[]$T, f: proc(T) -> bool) -> (index: int, f
...
+
+	index, found = linear_search_reverse(a, 0)
+	assert(index == -1 && found == false)
+	```
+
+	If you need a custom compare procedure, see `linear_search_reverse_proc`
+*/
+@(require_results)
+linear_search_reverse :: proc(array: $A/[]$T, key: T) -> (index: int, found: bool)
+	where intrinsics.type_is_comparable(T) #no_bounds_check {
```
Comment (laytan) [inline core/slice/slice.odin:163]: `#no_bounds_check` isn't needed, none of the operations add bounds checks (same thing with the reverse proc).

Comment (kawaii-Code) [reply core/slice/slice.odin:163]: What about plain `linear_search`? It does use `#no_bounds_check`

Comment (laytan) [reply core/slice/slice.odin:163]: I see, I am also pretty sure it isn't needed there.

**Comment by kawaii-Code** [comment]
@laytan done. There seem to be more `#no_bounds_check` scattered across `slice.odin`, e.g. in `filter_reverse`, `unique`, etc. Are you sure they aren't needed?

**Comment by laytan** [comment]
> @laytan done. There seem to be more `#no_bounds_check` scattered across `slice.odin`, e.g. in `filter_reverse`, `unique`, etc. Are you sure they aren't needed?

Yes, bounds checks are done when you slice or otherwise index into an array, in these search procs you just loop which afaik doesn't emit a bounds check.

**Pull request merged**